interp/cfg.go

```diff
@@ -644,7 +662,12 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 			}
 
 			switch n.action {
-			case aRem, aShl, aShr:
+			case aRem:
+				n.typ = c0.typ
+			case aShl, aShr:
+				if c0.typ.untyped {
+					break
+				}
```

Comment:
c0 is not necessarily a const here, It can be an untyped var i.e `1 << a` where `a` is a var. In that case we precisely want to use the context inherited type and not the one from child. The untyped criteria is the correct one.

Reply:
But in your example (`1 << a`) , `c0` is not `a`, since `c0` is the 0th child of the expression, and `a` is at the end of the expression? no?